src/bun.js/webcore/streams.zig

```diff
@@ -52,6 +52,48 @@ pub const ReadableStream = struct {
     value: JSValue,
     ptr: Source,
 
+    pub const Strong = struct {
+        held: JSC.Strong = .{},
+        globalThis: ?*JSGlobalObject = null,
+
+        pub fn init(this: ReadableStream, globalThis: *JSGlobalObject) !Strong {
+            switch (this.ptr) {
+                .Blob => |stream| {
+                    try stream.parent().incrementCount();
+                },
+                .File => |stream| {
+                    try stream.parent().incrementCount();
+                },
+                .Bytes => |stream| {
+                    try stream.parent().incrementCount();
+                },
+                else => {},
+            }
+            return .{
+                .globalThis = globalThis,
+                .held = JSC.Strong.create(this.value, globalThis),
+            };
+        }
+
+        pub fn get(this: *Strong) ?ReadableStream {
+            if (this.globalThis) |globalThis| {
+                if (this.held.get()) |value| {
+                    return ReadableStream.fromJS(value, globalThis);
+                }
+            }
+            return null;
+        }
+
+        pub fn deinit(this: *Strong) void {
+            if (this.get()) |readable| {
```

Comment:
This leaks memory in the case where the Strong's ptr exists but the value has been cleared from it. We can always call `this.held.deinit()` at the end.

Reply:
fixed in https://github.com/oven-sh/bun/pull/4418/commits/7cd66913e79f7c27d13efeb30801dfcbbabe45e6